src/rsvc/srv.c

```diff
@@ -410,6 +470,8 @@ rsvc_step_up_cb(struct rdb *db, uint64_t term, void *arg)
 	} else if (rc != 0) {
 		D_ERROR("%s: failed to step up as leader "DF_U64": %d\n",
 			svc->s_name, term, rc);
+		if (rsvc_class(svc->s_class)->sc_map_dist != NULL)
+			fini_map_distd(svc);
 		goto out_mutex;
```

Comment:
Yep, the same reasoning went through my head when I was writing this patch. I also felt it would be a bit of an overkill as there is only one place that needs to call fini_map_distd.